pkg/cmd/cli/cmd/create/clusterquota.go

```diff
@@ -0,0 +1,148 @@
+package create
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	quotaapi "github.com/openshift/origin/pkg/quota/api"
+)
+
+const (
+	ClusterQuotaRecommendedName = "clusterquota"
+
+	clusterQuotaLong = `
+Create a deployment config that uses a given image.
+
+Deployment configs define the template for a pod and manages deploying new images or configuration changes.`
+
+	clusterQuotaExample = `  # Create an nginx deployment config named my-nginx
+  %[1]s my-nginx --image=nginx`
+)
+
+type CreateClusterQuotaOptions struct {
+	ClusterQuota *quotaapi.ClusterResourceQuota
+	Client       client.ClusterResourceQuotasInterface
+
+	Mapper       meta.RESTMapper
+	OutputFormat string
+	Out          io.Writer
+	Printer      ObjectPrinter
+}
+
+// NewCmdCreateServiceAccount is a macro command to create a new service account
+func NewCmdCreateClusterQuota(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	o := &CreateClusterQuotaOptions{Out: out}
+
+	cmd := &cobra.Command{
+		Use:     name + " NAME --selector=key=value [resource=count]...",
+		Short:   "Create clusterquota resource.",
+		Long:    clusterQuotaLong,
+		Example: fmt.Sprintf(clusterQuotaExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(cmd, f, args))
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Run())
+		},
+		Aliases: []string{"dc"},
+	}
+
+	cmd.Flags().String("selector", "", "The project selector for the clusterquota")
```

Comment:
in case quota ever does support field selectors, should we instead keep this name specific to the kind it selects and make it --project-selector ?